File: careapp_mobile/lib/models/dependent.dart
class Dependent {
  final String id;
  final String fullName;
  final String relationship;
  final DateTime? dateOfBirth;
  final String? nationalId;
  final String? healthNotes;
  final String? specialNotes;
  
  Dependent({
    required this.id,
    required this.fullName,
    required this.relationship,
    this.dateOfBirth,
    this.nationalId,
    this.healthNotes,
    this.specialNotes,
  });
  
  factory Dependent.fromJson(Map<String, dynamic> json) {
    return Dependent(
      id: json['_id'] ?? json['id'] ?? '',
      fullName: json['fullName'] ?? '',
      relationship: json['relationship'] ?? '',
      dateOfBirth: json['dateOfBirth'] != null ? DateTime.parse(json['dateOfBirth']) : null,
      nationalId: json['nationalId'],
      healthNotes: json['healthNotes'],
      specialNotes: json['specialNotes'],
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'relationship': relationship,
      'dateOfBirth': dateOfBirth?.toIso8601String(),
      'nationalId': nationalId,
      'healthNotes': healthNotes,
      'specialNotes': specialNotes,
    };
  }
  
  int get age {
    if (dateOfBirth == null) return 0;
    final now = DateTime.now();
    int age = now.year - dateOfBirth!.year;
    if (now.month < dateOfBirth!.month || 
        (now.month == dateOfBirth!.month && now.day < dateOfBirth!.day)) {
      age--;
    }
    return age;
  }
}

File: careapp-backend/models/Dependent.js
const mongoose = require('mongoose');

const dependentSchema = new mongoose.Schema({
  clientId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Client',
    required: true
  },
  
  // ==================== Basic Information ====================
  fullName: { type: String, required: true },
  relationship: {
    type: String,
    required: true
  },
  gender: { type: String, enum: ['M', 'F'] },
  dateOfBirth: { type: Date, required: true },
  
  // ==================== Identifiers ====================
  nationalId: { type: String },
  age: { type: Number }, // calculated from dateOfBirth
  
  // ==================== Medical Information (text) ====================
  healthConditions: [{ type: String }],
  medications: [{ type: String }],
  allergies: [{ type: String }],
  healthNotes: { type: String },
  emergencyContact: {
    name: String,
    phone: String,
    relationship: String
  },
  
  // ==================== Link to separate MedicalInfo document (optional) ====================
  medicalInfoId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'MedicalInfo',
    default: null
  },
  
  // ==================== Status ====================
  isActive: { type: Boolean, default: true },
  status: {
    type: String,
    enum: ['active', 'inactive'],
    default: 'active'
  },
  
  // ==================== Timestamps ====================
  createdAt: { type: Date, default: Date.now },
  updatedAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Dependent', dependentSchema);